### PR TITLE
Form builder

### PR DESCRIPTION
**Pull request by jonasNobile**

### CODE REVIEW

File: forms/admin/components/ContentField.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		RB.headingFour,
+	],
+	[
+		RB.unorderedList,
+		RB.orderedList,
+	],
+	[
+		RB.code,
+		RB.anchor
+	]
```
Comment (VojtaStanek) [inline forms/admin/components/ContentField.tsx:35]:
```suggestion
		RB.anchor,
	],
```

File: forms/admin/components/PreviewLink.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+		return (
+			<a href={`${webUrl}${path}`} target="_blank">
+				{locale['Preview']}
+			</a>
+		)
+	},
+	({ slugField }) => (
+		<Field field={slugField} />
+	),
```
Comment (VojtaStanek) [inline forms/admin/components/PreviewLink.tsx:28]: Use FieldView instead

File: forms/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
+# Forms starter
+
+With this template we'll use Contember platform as headless form builder and [Next.js](https://nextjs.org/) for frontend site. You'll get:
+
+- Forms with 8 input blocks (Short answer, Long answer, Multiple choise, Check boxes, Drop down, File upload, Date, Date and time)
+- Responses for saving forms from the frontend website
```
Comment (VojtaStanek) [inline forms/README.md:6]:
```suggestion
- Saving form responses from the frontend website
```

File: forms/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+- [ACL rules](https://docs.contember.com/schema/acl) for administrators and public access for frontend website
```
Comment (VojtaStanek) [inline forms/README.md:7]:
```suggestion
- [Access control rules](https://docs.contember.com/schema/acl) for administrators and public access for frontend website
```

File: forms/admin/components/PreviewLink.tsx (hunk 2 above)
Comment (VojtaStanek) [reply forms/admin/components/PreviewLink.tsx:28]: Share it with `UrlCellFormat`

File: forms/admin/pages/forms.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	<DataGridPage
+		entities="Form"
+		itemsPerPage={50}
+		rendererProps={{ title: locale['Forms'], actions: <LinkButton to="formCreate">{locale['Add form']}</LinkButton> }}
+	>
+		<TextCell field="title" header={locale['Title']} />
+		<UrlCell field="slug" header={locale['Url']} prefix="form" />
+		<DateCell field="publishAt" header={locale['Publish date']} />
+		<TextCell field="details.responsesCount" header={locale['Number of responses']} />
+		<GenericCell shrunk canBeHidden={false}>
```
Comment (VojtaStanek) [inline forms/admin/pages/forms.tsx:17]: (We should add `ButtonCell` to core)

File: forms/api/acl.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+					},
+				},
+				Image: {
+					predicates: {},
+					operations: readOnly(model, 'Image', true),
+				},
+				Response: {
+					predicates: {},
+					operations: {
+						read: allField(model, 'Response', true),
```
Comment (VojtaStanek) [inline forms/api/acl.ts:86]: I thought that responses shouldn't be public...

File: forms/admin/forms/formForms.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+type FormSideFormProps = {
+	isEditPage?: boolean
+}
+
+export const FormSideForm = Component<FormSideFormProps>(
+	({ isEditPage }) => (
+		<>
+			{isEditPage &&
+				<PreviewLink slugField="slug" />
```
Comment (VojtaStanek) [inline forms/admin/forms/formForms.tsx:16]: Maybe check in the component if it exists and dont't use this prop?

File: forms/admin/forms/formForms.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+			}
+			<DateTimeField
+				field="publishAt"
+				label={locale['Publish date']}
+				defaultValue={new Date().toISOString()}
+			/>
+			<LinkButton to="responseList(id: $entity.id)" >{locale['See responses']}</LinkButton>
```
Comment (VojtaStanek) [inline forms/admin/forms/formForms.tsx:23]:
```suggestion
			<LinkButton to="responseList(id: $entity.id)">{locale['See responses']}</LinkButton>
```

File: forms/admin/pages/responses.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+		rendererProps={{ navigation: <NavigateBackButton to="responseList(id: $entity.form.id)">Responses</NavigateBackButton> }}
+	>
+		<Field field="createdAt" />
+		<br /><br />
+		<HasMany field="answers">
+			<Field field="input.question" />
+			<br />
+			<Field field="value" />
+			<br /><br />
+		</HasMany>
```
Comment (VojtaStanek) [inline forms/admin/pages/responses.tsx:35]: Does this look good? Maybe use `<dd>`, `<dt>`, `<dl>`...

File: forms/admin/pages/responses.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+			</LayoutPage>
+			<StaticRender>
+				<Field field="title" />
+			</StaticRender>
+		</EntitySubTree>
+	</DataBindingProvider>
+)
+
+export const responseView = (
+	<EditPage
```
Comment (VojtaStanek) [inline forms/admin/pages/responses.tsx:24]: is there save button shown? Shoudn't we hide it?

File: forms/api/model/Response.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+export class Response {
+	form = def.manyHasOne(Form, 'responses').notNull().cascadeOnDelete()
+	answers = def.oneHasMany(ResponseAnswer, 'response')
+	createdAt = def.dateTimeColumn().notNull().default('now')
+}
+
+export class ResponseAnswer {
+	value = def.stringColumn().notNull()
+	input = def.manyHasOne(FormInput).notNull()
+	response = def.manyHasOne(Response, 'answers').cascadeOnDelete()
```
Comment (VojtaStanek) [inline forms/api/model/Response.ts:14]:
```suggestion
	response = def.manyHasOne(Response, 'answers').cascadeOnDelete().notNull()
```

File: forms/head/components/header.tsx [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import Link from "./link"
+
+export default function Header(props: any) {
+	const { menu, logo } = props
```
Comment (VojtaStanek) [inline forms/head/components/header.tsx:4]:
```suggestion
export default function Header({ menu, logo }: any) {
```

File: forms/head/components/header.tsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+	return (
+		<header>
+			{logo &&
+				<img src={logo.url} width={logo.width} height={logo.height} alt={logo.alt} />
+			}
+			{menu &&
+				menu.items.map((item: any) => (
```
Comment (VojtaStanek) [inline forms/head/components/header.tsx:11]:
```suggestion
			{menu?.items.map((item: any) => (
```

File: forms/head/components/inputs.tsx [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+function MultipeChoice({ question, options, id: questionId }) {
+	return (
+		<fieldset key={questionId}>
+			<legend>{question}</legend>
+			{options?.map(({ id, value }) => (
+				<>
+					<label htmlFor={id}>
+						<input type="radio" id={id} key={id} name={questionId} value={value} />
```
Comment (VojtaStanek) [inline forms/head/components/inputs.tsx:26]:
```suggestion
					<label>
						<input type="radio" key={id} name={questionId} value={value} />
```

File: forms/admin/pages/forms.tsx (hunk 5 above)
Comment (jonasNobile) [reply forms/admin/pages/forms.tsx]: Added to our todo list.

**Pull request merged**